python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1201,3 +1202,26 @@ def __repr__(self) -> str:
     @classmethod
     def empty_subset(cls, partitions_def: PartitionsDefinition[T_cov]) -> "PartitionsSubset[T_cov]":
         return cls(partitions_def=partitions_def)
+
+
+class CachingPartitionsLoader:
+    def __init__(self):
+        self._partitions_by_def: Dict[PartitionsDefinition, Sequence[Partition]] = {}
+
+    def get_partition(
+        self,
+        partitions_def: PartitionsDefinition,
+        partition_key: str,
+        current_time: Optional[datetime] = None,
```

Comment:
This eseentially ignores current_time if the partitions_def is already in the cache, right? is that bad? Maybe not in practice? Not sure what exactly current_time is used for. Maybe it should go in the constructor of the object though?

Reply:
current_time is used so that time-based partitions know when their latest partition is (partitions only pop into existence after their time window is over).

within the context of tick, this shouldn't change (of course it's possible to have this happen, but it's rare and probably not worth worrying about, as if a partition exists at T_0, then it will exist at T_0+1).

not against putting it in the constructor, but don't think it makes a big difference for now, as in general the current_time that's passed in here seems to be static throughout the duration of the tick anyway

Reply:
`current_time` is used to determine which time window partitions are existent at the given time. Within the context of a tick, this won't change.

But i've gone ahead and moved these args to the constructor. Sensors will still pass in `current_time=None` since they don't have a notion about current time, but I added comments to clarify that the partition loader is intended to be used within a tick (because the partitions don't change during the course of a tick).